lesson11/main.js

```diff
@@ -0,0 +1,71 @@
+const ul = document.getElementById("js-ul");
+
+const addLoading = () =>{
+  const loading = document.getElementById("js-loading");
+  const loadingImg = document.createElement('img');
+  loadingImg.src = 'loading.gif';
+  loadingImg.id = "loading-img";
+  loading.appendChild(loadingImg);
+};
+
+const removeLoading = () =>{
+  document.getElementById('loading-img').remove();
+};
+
+const renderLists = (features) =>{
+  const fragment = new DocumentFragment();
+
+  for (let feature of features ){
+      const li = document.createElement('li');
+      const a = document.createElement('a');
+      const img = document.createElement('img');
+      a.href = feature.to;
+      a.textContent = feature.text;
+      img.src = feature.img;
+      img.alt = feature.alt;
+
+      fragment.appendChild(li).appendChild(a).insertAdjacentElement('afterbegin', img);
+    };
+
+    ul.appendChild(fragment);
+};
+
+const displayError = (error) =>{
+  const li = document.createElement('li');
+  li.textContent = error;
+  ul.appendChild(li);
+}
+
+
+const getResponse = new Promise((resolve, reject) => {
+  return fetch('http://myjson.dit.upm.es/api/bins/dg7g').then(response => {
```

Comment:
MUST
You need to change the URL from http to https because CodeSandBox will give you an error.

Reply:
Thank you for your feedback Mai-san,

I removed the original new Promise() and refactored some code.
Please check my latest commit 9ce0dd07ed5047f64e1e47248c5d9b0f53dd8142